pkg/image/qemu.go

```diff
@@ -259,3 +259,17 @@ func (o *qemuOperations) CreateBlankImage(dest string, size resource.Quantity, p
 
 	return nil
 }
+
+// PreallocateBlankBlock writes requested amount of zeros to block device mounted at dest
+func PreallocateBlankBlock(dest string, size resource.Quantity) error {
+	klog.V(3).Infof("block volume size is %s", size.String())
+
+	args := []string{"if=/dev/zero", "of=" + dest, "bs=" + convertQuantityToQemuSize(size), "count=1"}
```

Comment:
Any particular reason you are not calling CreateBlankDisk with a preallocate=full?  I noticed we already added a preallocation=falloc, why not just allow full with a different value parameter?

Reply:
What's `CreateBlankDisk`? There doesn't seem to be anything like that in the code.

If you meant `CreateBlankImage` – do we want to use qemu-img at this case? It's just a raw block device. Also, quickly checking, it seems that qemu-img simply does not support it, at least for some storage (local in this case):

```
> cluster-up/ssh.sh node01 "sudo qemu-img create -f raw /dev/loop0 100M -o preallocation=falloc"
2021/01/15 14:41:05 Waiting for host: 192.168.66.101:22
2021/01/15 14:41:05 Connected to tcp://192.168.66.101:22
qemu-img: /dev/loop0: Unsupported preallocation mode 'falloc'
Formatting '/dev/loop0', fmt=raw size=104857600 preallocation=falloc

> cluster-up/ssh.sh node01 "sudo qemu-img create -f raw /dev/loop0 100M -o preallocation=full"
2021/01/15 14:41:08 Waiting for host: 192.168.66.101:22
2021/01/15 14:41:08 Connected to tcp://192.168.66.101:22
Formatting '/dev/loop0', fmt=raw size=104857600 preallocation=full
qemu-img: /dev/loop0: Unsupported preallocation mode 'full'

```

Am I missing something?

Reply:
No you are correct, you can't create a block device, it exists already, so we need to dd.